[CIAM-535] [KEYCLOAK-18574] [CIAM-690] [CIAM-696] [CIAM-711] [CIAM-743] Initial RH-SSO 7.5.0.GA image

iankko opened the pull request:
[CIAM-535] [KEYCLOAK-18574] [CIAM-690] [CIAM-696] [CIAM-711] [CIAM-743] Initial RH-SSO 7.5.0.GA image

See the particular commits (they have been intentionally kept separated) for additional details about the underlying changes

Corresponding pipeline runs:

Image build / testing: https://keycloak.jenkins.server.com/view/RH-SSO/job/rh-sso-pipeline-continuous_container/205/

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

Commit:
[CIAM-535] [KEYCLOAK-18574] Refactor the Red Hat Single Sign-On for OpenShift
images to start using Wildfly / JBoss EAP Galleon based Maven repository
instead of a formerly used server Zip archive. Update to RH-SSO 7.5.0

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
@drichtarik jFYI:

- There are hardcoded MD5 sums of `maven-repo.zip` & `keycloak-server-overlay.zip` artifacts, 
- And the `keycloak.openshift.clients` module expects the Zip artifacts with the adapters to be prefixed with the `keycloak-` prefix (rather than by the `rh-sso` one, expected by CPaaS)

so the image is buildable using the old pipeline too. But those two points are just easily fixable (for the case of CPaaS build) nuances.

The other changes would apply for CPaaS pipeline too.

Commit:
[CIAM-690] keycloak.sh -- Add support for RH-SSO 7.5.0,
when retrieving PEM certificate of the signing key of
the realm

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[CIAM-696] Replace Bash "read -r -d '' variable <<- Here
String" constructs with the "variable=$(cat <-- Here String)" ones, since the
former doesn't work correctly with the Bash "set -e" setting, which is
simultaneously set on RH-SSO image modules scripts. This fixes the RH-SSO
7.5-1's image's failure to start when JGROUPS_ENCRYPT_PROTOCOL variable is
set to ASYM_ENCRYPT protocol

Also add vim-minimal RPM to the list of RPMs installed in the image
(handy for debugging)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

pskopek commented on modules/eap/setup/eap/modules/module.yaml:
```diff
@@ -17,11 +17,11 @@ packages:
     - which
-    # "tree" tool is handy to inspect directory structure when needed
     - tree
+    - vim-minimal
```
Is vim really necessary here? We should keep number of pkgs as low as possible.

iankko replied:
Not really. It's handy, but can be installed each time, when the container is run (with `-u root` args). Dropped it from latest version

Commit:
[CIAM-711] Update RH-SSO server overlay Zip artifact to
'RH-SSO 7.5.0.CR4' build

Also add a note to 'modules.eap' module, about the logic, which is
used for the inclusion of necessary EAP modules into the RH-SSO modules
(since the approach changed due to OSBS container build system
limitations)

Drop vim-minimal RPM package

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Commit:
[CIAM-743] Explicitly add "org.keycloak.keycloak-model-infinispan"
module as a dependency to the definition of the Wildfly's
"org.wildfly.clustering.infinispan.spi" module.xml

Rationale:
----------

The "org.wildfly.clustering.infinispan.spi" module of vanilla Wildfly / EAP
doesn't deploy RH-SSO server subsystem on top of Wildfly / EAP (thus doesn't
require this dependency).

But since the RH-SSO server overlay does, the "org.wildfly.clustering.infinispan.spi"
requires explicitly to be equipped also with the "org.keycloak.keycloak-model-infinispan"
module dependency, so the classes, defined within this RH-SSO module are properly found,
when referenced

Besides that, the JBoss EAP 7.4.0 image builder Maven repository got moved from
candidates/ to released/. Update the path to that artifact appropriately

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

pskopek reviewed (approved):
used in successful build of rh-sso 7.5.0
